sphynx/lynxkite-sphynx/networkit_compute_attribute.go

```diff
@@ -0,0 +1,41 @@
+// All NetworKit ops that compute a vertex attribute on a graph.
+package main
+
+import (
+	"github.com/lynxkite/lynxkite/sphynx/networkit"
+)
+
+func ToNetworKit(vs *VertexSet, es *EdgeBundle) networkit.Graph {
+	builder := networkit.NewGraphBuilder(uint64(len(vs.MappingToUnordered)))
+	for i := range es.Src {
+		builder.AddHalfEdge(uint64(es.Src[i]), uint64(es.Dst[i]))
+	}
+	return builder.ToGraph(true)
+}
+
+func init() {
+	operationRepository["NetworKitComputeAttribute"] = Operation{
+		execute: func(ea *EntityAccessor) error {
+			vs := ea.getVertexSet("vs")
+			es := ea.getEdgeBundle("es")
+			g := ToNetworKit(vs, es)
+			attr := &DoubleAttribute{
+				Values:  make([]float64, len(vs.MappingToUnordered)),
+				Defined: make([]bool, len(vs.MappingToUnordered)),
+			}
+			var result networkit.DoubleVector
+			switch ea.GetStringParam("op") {
+			case "betweenness":
+				b := networkit.NewBetweenness(g)
+				b.Run()
+				result = b.Scores()
+			}
+			attr.Values = networkit.ToDoubleSlice(result)
+			for i := range attr.Defined {
+				attr.Defined[i] = true
+			}
```

Comment:
Oh, it's been a while since I looked at go code. Really, this is the way you fill a slice with a constant value? *Sigh*

Reply:
> Oh, it's been a while since I looked at go code. Really, this is the way you fill a slice with a constant value? _Sigh_

![Screenshot 2020-11-17 at 10 49 03](https://user-images.githubusercontent.com/1268018/99374340-c2afce00-28c2-11eb-8fbc-089a2bd13d28.png)